index.js

```diff
@@ -33,23 +33,23 @@ class ProfileData {
 
      
 
-    passRender(){
+    renderPass() {
         profile.innerHTML += 
              `<div class = "stampContainer">
                 <p class = "passStamp">NOPE</p>
               </div>`
     }
 }
 let doggy = {}
+let index = 0;
 
 function getNewDoggy(){
-    const nextDoggy = dogs[dogsNames.shift()]
+    const nextDoggy = dogs[dogsNames[index % dogsNames.length]];
+    index++;
     console.log(nextDoggy) 
-    doggy = nextDoggy ? new ProfileData(nextDoggy) : {}
-    profile.innerHTML = doggy.renderProfile()
-    console.log(dogsNames.length)
-    if(dogsNames.length === 0){
-        dogsNames = ["Rex",'Bella','Teddy']
+    if (nextDoggy) { 
+        doggy = new ProfileData(nextDoggy);
+        profile.innerHTML = doggy.renderProfile();
```

Comment:
if you will try to trigger method 'renderProfile' from {} empty object you'll get an error. 

PS: Especially if someone remove one of dogs data from your data.js file and will not update an array of name in 2!!!! places in code.